addons/web/static/src/js/core/ajax.js

```diff
@@ -17,9 +17,17 @@ function genericJsonRpc (fct_name, params, settings, fct) {
         params: params,
         id: Math.floor(Math.random() * 1000 * 1000 * 1000)
     };
+    var start_time = Math.floor(Date.now());
     var xhr = fct(data);
     var result = xhr.pipe(function(result) {
-        core.bus.trigger('rpc:result', data, result);
+        var end_time = Math.floor(Date.now());
+        var timings = {
+            'method': url || params['method'],
+            'request_time': (end_time-start_time) / 1000,
+            'server_time': parseFloat(xhr.getResponseHeader('x-server-time'))
+        };
+        core.bus.trigger('rpc:result', data, result, timings);    
```

Comment:
Not familiar with this call. I think it's meant to enable comms between modules. But I assume that it will handle and propagate the extra argument, right?

Reply:
yes